mffpy/writer.py

```diff
@@ -34,25 +34,31 @@ def __init__(self, filename: str):
         self.filename = filename
         self.files: Dict[str, Any] = {}
         self._bin_file_added = False
+        self.mffdir, self.ext = splitext(self.filename)
+        self.mffdir += '.mff'
+        self.file_created = False
+
+    def create_directory(self):
+        if not self.file_created:
+            makedirs(self.mffdir, exist_ok=False)
+            self.file_created = True
 
     def write(self):
         """write contents to .mff/.mfz file"""
-        # create .mff directory
-        mffdir, ext = splitext(self.filename)
-        mffdir += '.mff'
-        makedirs(mffdir, exist_ok=False)
+
+        self.create_directory()
```

Comment:
Not disagreeing, but why break out this functionality into its own function?

Reply:
Because currently it only creates the folder when write is called. Write needs to be called only after all of the data is collected, but in order to stream the EEG data to a file, the directory needs to already exist.